nixos/modules/services/games/pterodactyl-wings.nix

```diff
@@ -0,0 +1,214 @@
+                description = "Path to certificate to use for HTTPS.";
+              };
+              key = mkOption {
+                type = nullOr path;
+                default = null;
+                description = "Path to key to use for HTTPS.";
+              };
+            };
+          };
+          system = {
+            root_directory = mkOption {
+              type = path;
+              default = "/var/lib/pterodactyl";
+              description = "Root directory where daemon data is stored.";
+            };
+            log_directory = mkOption {
+              type = path;
+              default = "/var/log/pterodactyl";
+              description = "Directory where logs for server installations and other wings events are logged.";
+            };
+            data = mkOption {
+              type = path;
+              defaultText = "/var/lib/pterodactyl/volumes";
+              description = "Directory where game server data is stored.";
+            };
+            archive_directory = mkOption {
+              type = path;
+              defaultText = "/var/lib/pterodactyl/archives";
+              description = "Directory where server archives for cross-node transferring will be stored.";
+            };
+            backup_directory = mkOption {
+              type = path;
+              defaultText = "/var/lib/pterodactyl/backups";
+              description = "Directory where local backups will be stored.";
+            };
+            username = mkOption {
+              type = str;
+              default = "pterodactyl";
+              description = ''
+                The user that should own all of the server files, and be used for containers.
+                NixOS will create this user and group if left at the default.
+              '';
+            };
+            sftp = {
+              bind_address = mkOption {
+                type = str;
+                default = "0.0.0.0";
+                example = "[::1]";
+                description = "The bind address of the SFTP server.";
+              };
+              bind_port = mkOption {
+                type = port;
+                default = 8080;
+                description = "The bind port of the SFTP server.";
+              };
+              read_only = mkOption {
+                type = bool;
+                default = false;
+                description = "If set to true, no write actions will be allowed on the SFTP server.";
+              };
+            };
+          };
+          docker = {
+            network = {
+              interface = mkOption {
+                type = str;
+                default = "172.18.0.1";
+                description = "The interface that should be used to create the network. Must not conflict with any other interfaces in use by Docker or on the system.";
+              };
+              dns = mkOption {
+                type = listOf str;
+                default = [ "1.1.1.1" "1.0.0.1" ];
```

Comment:
Why not default to the hosts DNS settings?

Reply:
I wasn't sure how to implement that, so I left it as the defaults of the configuration file. I thought of using `networking.nameservers`, but it's empty by default because most will probably use DHCP. Any way to evaluate the host's servers in the module?